### PR TITLE
Add support for qualified interface access

### PR DESCRIPTION
**Pull request by DanilaFe**
Closes https://github.com/chapel-lang/chapel/issues/25829.

Strangely, even though we have an `isym` when building the class hierarchy, I chose to use the symbol's name to construct an `implements` statement. This works when the name is available in scope without qualification, but not when qualified access is used. The principled solution is to allow interface statements to be constructed using a known interface symbol. This PR does that, and switches the class hierarchy logic to avoid using a name.

Reviewed by @jabraham17 -- thanks!

## Testing
- [x] paratest

### CODE REVIEW

File: test/constrained-generics/basic/qualified-interface-constraint.chpl [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module Other {
+  class CC {}
+  interface II {}
+}
+
+module M {
+  import Other;
+  class C: Other.CC {}
+  record R: Other.II {}
```
Comment (jabraham17) [inline test/constrained-generics/basic/qualified-interface-constraint.chpl:9]: What about this case?

```
record RR {}
RR implements Other.II;
```

File: compiler/include/stmt.h [hunk 2]
```diff
@@ -317,6 +317,9 @@ struct Witnesses {
 
 class ImplementsStmt final : public Stmt {
 public:
+  static ImplementsStmt* build(InterfaceSymbol* isym,
+                               CallExpr* actuals,
+                               BlockStmt* body);
   static ImplementsStmt* build(const char* name,
```
Comment (jabraham17) [inline compiler/include/stmt.h:323]: do we still need the `const char*` overload? Is it ever the correct thing to use?

 It is still used by `convert-uast`, but I suspect that to make `R implements Other.I` work that also should change.

Comment (DanilaFe) [reply compiler/include/stmt.h:323]: Well, we don't always have scope resolution results available in `convert-uast.cpp`, so `const char*` will be needed sometimes. I can adjust it to use the interface symbol when possible.

File: test/constrained-generics/basic/qualified-interface-constraint.chpl (hunk 1 above)
Comment (jabraham17) [reply test/constrained-generics/basic/qualified-interface-constraint.chpl:9]: you are correct, it is not valid syntax

**Comment by DanilaFe** [comment]
Per an offline conversation with Jade, given that `implements` is unstable and that the `implements I.x` is not even valid syntax right now, this PR will not include that functionality.

**Pull request merged**